torch/nn/modules/conv.py

```diff
@@ -610,22 +610,24 @@ def __init__(self, in_channels, out_channels, kernel_size, stride,
     # compatibility
     def _output_padding(self, input: Tensor, output_size: Optional[List[int]],
                         stride: List[int], padding: List[int], kernel_size: List[int],
-                        dilation: Optional[List[int]] = None) -> List[int]:
+                        num_spatial_dims: int, dilation: Optional[List[int]] = None) -> List[int]:
         if output_size is None:
             ret = _single(self.output_padding)  # converting to list if was not already
         else:
-            k = input.dim() - 2
-            if len(output_size) == k + 2:
-                output_size = output_size[2:]
+            has_batch_dim = input.dim() == num_spatial_dims + 2
+            num_non_spatial_dims = 2 if has_batch_dim else 1
+            k = num_spatial_dims
```

Comment:
Why use `k` here? Can't we just keep using `num_spatial_dims` ?

Reply:
You're right :) I was being lazy to minimize changes. Will fix